### PR TITLE
Bugfix // Android Manual Purchase Tracking

### PR DESCRIPTION
**Pull request by iVirn**
#### Changes
* Pass Android purchase info as JSON String
* Update Example app to use real purchase service

### CODE REVIEW

File: android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionFlutterSdkPlugin.kt [hunk 1]
```diff
@@ -111,14 +109,11 @@ class QonversionFlutterSdkPlugin internal constructor(registrar: Registrar): Met
         result.success(null)
     }
 
-    private fun createSkuDetails(map: Map<String, Any>): SkuDetails {
-        val json = map.toString()
+    private fun createSkuDetails(json: String): SkuDetails {
```
Comment (suriksarkisyan) [inline android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionFlutterSdkPlugin.kt:112]: Looks like this two functions can be removed and you can use `SkuDetails(json)` directly

File: example/lib/main.dart [hunk 2]
```diff
@@ -68,14 +84,83 @@ class _MyAppState extends State<MyApp> {
...
+
   ProductDetails getProductDetails() {
     // ignore: invalid_use_of_visible_for_testing_member
     final original = SkuDetailsWrapper(
       description: 'description',
       freeTrialPeriod: 'freeTrialPeriod',
       introductoryPrice: 'introductoryPrice',
       introductoryPriceMicros: 'introductoryPriceMicros',
-      introductoryPriceCycles: 'introductoryPriceCycles',
+      introductoryPriceCycles: null,
```
Comment (suriksarkisyan) [inline example/lib/main.dart:163]: Are you sure this correct?
That is how it works in Android SDK
`return if (details.freeTrialPeriod.isNullOrEmpty()) details.introductoryPriceCycles else 0`

Comment (iVirn) [reply example/lib/main.dart:163]: that's just a mock to test if parsing null values works correctly. doesn't make any difference here

Comment (suriksarkisyan) [reply example/lib/main.dart:163]: Oh, my bad, sorry 🙃

**Review by suriksarkisyan — APPROVED** [review]

**Pull request merged**